server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/TextCalendarBodyToAttachmentTest.java

```diff
@@ -0,0 +1,237 @@
+
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldKeepMessageAsItIsWhenMessageIsNotTextCalendar() throws Exception {
+        String messageContent = "Content-type: text/html; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getMessage().isMimeType("text/*")).isTrue();
+        assertThat(mail.getMessage().getDisposition()).isNull();
+    }
+
+    @Test
+    public void serviceShouldChangeMessageContentTypeToMultipart() throws Exception {
+        String messageContent = "Content-type: text/calendar; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getMessage().isMimeType("multipart/*")).isTrue();
+    }
+
+    @Test
+    public void serviceShouldConvertTextBodyOfMessageToAttachmentWhenTextCalendar() throws Exception {
+        String messageContent = "Content-type: text/calendar; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR\n" +
+            "END:VEVENT\n" +
+            "END:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        MimeMessage actual = mail.getMessage();
+        Multipart multipart = (Multipart)actual.getContent();
+
+        assertThat(multipart.getCount()).isEqualTo(1);
+        assertThat(multipart.getBodyPart(0).getDisposition()).isEqualTo("attachment");
+    }
+
+    @Test
+    public void serviceShouldKeepAllMessageHeaderWhenConverting() throws Exception {
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(MESSAGE_CONTENT.getBytes()));
+        List<Header> allHeaders = Collections.list(message.getAllHeaders());
+        List<String> allStringHeaders = allHeaders.stream()
+            .map(Header::getName)
+            .collect(Guavate.toImmutableList());
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        List<Header> actualHeaders = Collections.list(mail.getMessage().getAllHeaders());
```

Comment:
Please no .stream in your tests. Show explicitly expected header values.

Reply:
There are some header's value are changed like Message-ID, Content-Type  so I cannot assert them all

Reply:
MessageId IMO should not change IMO.

Reply:
Asset header by hender please...

Reply:
I meant:

```
assertThat(actualHeader)
   .extracting(Header::getName, Header::getValue)
   .contains(
    Tuple.tuple("Return-Path", "<local@localhost.com>"),
    Tuple.tuple("MIME-Version", "1.0"),
    Tuple.tuple("Delivered-To", "to@localhost.com"),
    Tuple.tuple("Message-ID", "<2d0fa16c-4c37-58de-1acc-50b5f6a23a60@localhost.com>"),
    Tuple.tuple("Subject", "Any subject"),
    ...);
```

instead of that uggly for loop. *Your assertions should be explicit.*

And actually my original comment was focussed on **Body part headers** (Content-Type, Content-Transfer-Encoding)... I don't believe this test is really useful...

Reply:
I wondered that also then I removed this UT. Noted for assert with list. Thx